src/main/controllers/features/limited/correspondenceAddressConfirmController.ts

```diff
@@ -2,37 +2,49 @@ import { NextFunction, Request, Response } from "express";
 import * as config from "../../../config";
 import { selectLang, addLangToUrl, getLocalesService, getLocaleInfo } from "../../../utils/localise";
 import { LIMITED_CORRESPONDENCE_ADDRESS_CONFIRM, LIMITED_CORRESPONDENCE_ADDRESS_MANUAL, LIMITED_CORRESPONDENCE_ADDRESS_LOOKUP, BASE_URL, LIMITED_SECTOR_YOU_WORK_IN } from "../../../types/pageURL";
-import { ACSPData } from "../../../model/ACSPData";
 import { Session } from "@companieshouse/node-session-handler";
-import { ANSWER_DATA, LIMITED_CORRESPONDENCE_ADDRESS, USER_DATA } from "../../../common/__utils/constants";
+import { ANSWER_DATA, USER_DATA, SUBMISSION_ID, GET_ACSP_REGISTRATION_DETAILS_ERROR } from "../../../common/__utils/constants";
 import { Answers } from "../../../model/Answers";
 import { saveDataInSession } from "../../../common/__utils/sessionHelper";
+import logger from "../../../../../lib/Logger";
+import { ErrorService } from "../../../services/errorService";
+import { getAcspRegistration } from "../../../services/acspRegistrationService";
+import { AcspData } from "@companieshouse/api-sdk-node/dist/services/acsp";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
     const lang = selectLang(req.query.lang);
     const locales = getLocalesService();
     const session: Session = req.session as any as Session;
-    const acspData: ACSPData = session?.getExtraData(USER_DATA)!;
+    const previousPage: string = addLangToUrl(BASE_URL + LIMITED_CORRESPONDENCE_ADDRESS_LOOKUP, lang);
+    const currentUrl: string = BASE_URL + LIMITED_CORRESPONDENCE_ADDRESS_CONFIRM;
 
-    res.render(config.CORRESPONDENCE_ADDRESS_CONFIRM, {
-        previousPage: addLangToUrl(BASE_URL + LIMITED_CORRESPONDENCE_ADDRESS_LOOKUP, lang),
-        editPage: addLangToUrl(BASE_URL + LIMITED_CORRESPONDENCE_ADDRESS_MANUAL, lang),
-        title: "Confirm the correspondence address",
-        ...getLocaleInfo(locales, lang),
-        currentUrl: BASE_URL + LIMITED_CORRESPONDENCE_ADDRESS_CONFIRM,
-        businessName: acspData?.businessName,
-        correspondenceAddress: acspData?.address
+    try {
+        // get data from mongo
+        const acspData = await getAcspRegistration(session, session.getExtraData(SUBMISSION_ID)!, res.locals.userId);
```

Comment:
Forgot to save in session variable.

Reply:
Purposefully skipped it as I am not updating the object and saving in DB on this page. Correspondence Address are filled up in previous pages.